packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -454,31 +457,46 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			0x385 /* For clarity and paranoid defense against adding future attachment states */,
 		);
 
+		if (signal?.aborted) {
+			throw(Error("aborted before uploading"));//signal as any).reason));
+		}
+
 		// Create a local ID for the blob. After uploading it to storage and before returning it, a local ID to
 		// storage ID mapping is created.
 		const localId = uuid();
 		const pendingEntry: PendingBlob = {
 			blob,
 			status: PendingBlobStatus.OnlinePendingUpload,
 			handleP: new Deferred(),
-			uploadP: this.uploadBlob(localId, blob),
+			uploadP: this.uploadBlob(localId, blob, signal),
 			attached: false,
 			acked: false,
```

Comment:
you should add a test, and handle the case where the upload is canceled while waiting on the op, it looks like that will assert currently.

Reply:
also need to test, resbmit of op for canceled upload

Reply:
I added the test for waiting on the op. Not sure how to test resubmit though. By reconnecting after uploading the blob?

Reply:
i believe there is a mock runtime that supports reconnect that could be used.

Reply:
it looks like the blob manager tests have their own mock runtime, you should extend, or build your own to exercise all the functionality here.